src/main/java/net/imglib2/algorithm/hough/HoughTransforms.java

```diff
@@ -0,0 +1,343 @@
+	 * the image origin to the line, and {@code theta} indicates the angle
+	 * between the row-axis and the minimum offset vector.
+	 * </p>
+	 * <p>
+	 * For a given point, then, votes are placed along the curve
+	 * </p>
+	 * 
+	 * <pre>
+	 * rho = y * sin( theta ) + x * cos( theta )
+	 * </pre>
+	 * <p>
+	 * It is important to note that the interval of the first dimension of the
+	 * vote space image is NOT {@code [-maxRho, maxRho)} but instead
+	 * {@code [0, maxRho * 2)}. Thus if {@link HoughTransforms#pickPeaks} is not
+	 * used to retrieve the maxima from the vote space, the vote space will have
+	 * to be translated by {@code -maxRho} in dimension 0 to get the correct
+	 * {@code rho} and {@code theta} values from the vote space.
+	 * </p>
+	 * 
+	 * @param input
+	 *            - the {@link RandomAccessibleInterval} to run the Hough Line
+	 *            Transform over
+	 * @param votespace
+	 *            - the {@link RandomAccessibleInterval} in which the results
+	 *            are stored
+	 * @param nRho
+	 *            - the number of bins for rho resolution
+	 * @param nTheta
+	 *            - the number of bins for theta resolution
+	 * @param threshold
+	 *            - the minimum value allowed by the populator. Any input less
+	 *            than or equal to this value will be disregarded by the
+	 *            populator.
+	 */
+	public static < T extends RealType< T > > void voteLines( final RandomAccessibleInterval< T > input, final RandomAccessibleInterval< T > votespace, final int nRho, final int nTheta, final T threshold )
+	{
+
+		final long[] dims = new long[ input.numDimensions() ];
+		input.dimensions( dims );
+
+		final double minRho = -computeLength( dims );
+		final double dRho = 2 * computeLength( dims ) / ( double ) nRho;
+
+		final double minTheta = -Math.PI / 2;
+		final double dTheta = Math.PI / nTheta;
+
+		final double[] rho = new double[ nRho ];
+		final double[] theta = new double[ nTheta ];
+
+		// create theta LUT
+		for ( int t = 0; t < nTheta; ++t )
+		{
+			theta[ t ] = dTheta * t + minTheta;
+		}
+
+		// create rho LUT
+		for ( int r = 0; r < nRho; ++r )
+		{
+			rho[ r ] = dRho * r + minRho;
+		}
+
+		final Cursor< T > imageCursor = Views.iterable( input ).localizingCursor();
+		final RandomAccess< T > outputRA = votespace.randomAccess();
+
+		final long[] position = new long[ input.numDimensions() ];
+
+		final T one = Util.getTypeFromInterval( input ).createVariable();
+		one.setOne();
+
+		while ( imageCursor.hasNext() )
+		{
+			double fRho;
+			int r;
+			final int[] voteLoc = new int[ 2 ];
+
+			imageCursor.fwd();
+			imageCursor.localize( position );
+
+			for ( int t = 0; t < nTheta; ++t )
+			{
+				if ( imageCursor.get().compareTo( threshold ) > 0 )
+				{
+					fRho = Math.cos( theta[ t ] ) * position[ 0 ] + Math.sin( theta[ t ] ) * position[ 1 ];
+					r = Math.round( ( float ) ( ( fRho - minRho ) / dRho ) );
+					voteLoc[ 0 ] = r;
```

Comment:
Use 
```
outputRA.setPosition( r, 0 );
outputRA.setPosition( rt 1 );
```
instead of writing into `voteLoc` and then copying into `outputRA`.
This will also make `voteLoc` obsolete (no need to create an int[] for every pixel of `input`)